### ISSUE TITLE
DDC control does not work on M2 Pro/Max systems

### ISSUE THREAD

**Issue by YumingNiu** [MonitorControl/MonitorControl#1301]
### Before opening the issue, have you...?

- [X] Searched for existing issues
- [X] Read through [the Monitor-Troubleshooting Wiki](https://github.com/MonitorControl/MonitorControl/wiki/Monitor-Troubleshooting)
- [X] Updated MonitorControl to the latest version (if applicable)

### Describe the issue

It cannot control the monitor hardware settings on my **M2 Pro Mac mini**.
Specifically, the volume slider and the contrast slider do not show up in the menu bar. Adjusting the brightness only adjusts the software brightness. When the sound output is set to the external display, adjusting the volume does not modify the volume although the macOS OSD changes.

LG's official app, _OnScreen Control_ does NOT work.
The other app, _BetterDisplay_, works well.

All three apps work on an Intel Macbook Pro with the same monitor.

### Expected behavior

Adjusting the volume and brightness should change the hardware settings as shown in the monitor's OSD.

### Environment Information (please complete the following information)

```markdown
- macOS version: 13.2 (22D49)
- Mac model: MNH73VC/A (Mac mini)
- MonitorControl version: 4.1.0
- Monitor(s): LG 27UP850N-W
- Monitor Cable(s)/Connection(s): CtoC
- Apple Silicon/M1 (yes or no): yes (M2 Pro)
```

**Comment by waydabber** [comment]
Due to various hardware changes, MonitorControl 4.1.0 is incompatible with M2 Pro/Max Macs (and M1 Macs with the built-in HDMI port). There are plans to update the app (+ of course it's open to any contributor to work on it meanwhile).

Until support is added, you can follow this: https://github.com/MonitorControl/MonitorControl/discussions/1299 (as I see you already did).

**Comment by vukvukovich** [comment]
@waydabber Hi, thanks for the great software!

For me it worked fine with M1 Mac till recently. Now I can't control volume with neither apps. Any ideas?

**Comment by danielparas** [comment]
+1 on M2

**Comment by waydabber** [comment]
Hi @vukvukovich - I am not sure why this could be the case. Does hardware brightness control (by default the 50-100% part of the brightness slider) work? M1 Macs with USB-C should be supported with MonitorControl (note: since your comment about M1, it feels a bit off-topic here. :)).

**Comment by vukvukovich** [comment]
@waydabber Sure, do you want me to create another ticket? 

Btw. Brightness works, also the volume sliders work in the terms of OSD but the volume itself stays the same.

**Comment by waydabber** [comment]
@vukvukovich well if brightness control works then DDC commands are getting through. It might be a display issue and I am afraid we can't do much about it. If volume control worked earlier with the display then maybe resetting the display will help to restore that state? I am not sure what the problem is though.

**Comment by vukvukovich** [comment]
Thanks, I'll try it out.

**Comment by salayidin** [comment]
hope it. can be fixed on M2 Pro /M2 Max ,really love this app

**Comment by carlos-algms** [comment]
Same here, the App still works after upgrading to the M2 Max, but only via "**Software (Gamma)**"

![image](https://user-images.githubusercontent.com/4634613/228461957-a5972e0d-497f-40d2-a3f7-b150b61d1d6e.png)

Looking forward to see this amazing App working on the newest chips.

**Comment by SiegfredRodriguez** [comment]
Experiencing this too with mac mini m2 pro, brightness is ok but volume is not.

**Comment by bradtchapman** [comment]
M2 MBP user reporting in with the same issue.  Hardware control works fine on my M1; not on the M2 that my company sent to me.

**Comment by evertonspader-tomtom** [comment]
Same issue here. Macbook M1 Pro works fine for volume and brightness but on M2 Pro only brightness is working

**Comment by cpwillis** [comment]
Same issue here. Since switching to M2 (from M1), brightness can only be controlled via "Software (Gamma)".

**Comment by sobimor** [comment]
Same issue here M2 Max

**Comment by CharlesCCC** [comment]
same issue here. (3 exactly the same monitors). My M1 Max works totally fine, and then the M2 Max doesn't. (Originally I thought it was caused by displaylink drive, but apparently wasn't the case).

**Comment by waydabber** [comment]
Honestly, I think it makes more sense to use BetterDisplay since it has more features and all DDC stuff is free in it, but I understand many like the simplicity and opensource nature of MonitorControl so I added this PR to cover M2 Pro/Max/Ultra systems:

https://github.com/MonitorControl/MonitorControl/pull/1404

**Comment by chanelxoxo** [comment]
Same issue here, M2 Max + Dell U2718Q + HDMI to HDMI (2.1) or DP 1.2 to USB-C cable.

**Comment by waydabber** [comment]
This is now resolved with the v4.2.0 update which contains the M2 Pro DDC additions carried over from BetterDisplay.

**Comment by CharlesCCC** [comment]
thank you @waydabber to get this resolved.

**Comment by working-name** [comment]
> This is now resolved with the v4.2.0 update which contains the M2 Pro DDC additions carried over from BetterDisplay.

Hi, just tried 4.3.2 and 4.3.3, neither version shows volume. Greyed out in options. I'm interested in smooth sound volume changes mainly, don't have a use for any of the display adjustments.

- macOS version: 14.7
- Mac model: Z160000B2LL/A (MacBook Air M2)
- MonitorControl version: 4.3.2/3
- Monitor(s): Built-in
- Monitor Cable(s)/Connection(s): beats me
- Apple Silicon/M1 (yes or no): yes (M2)
